pulsar-client-cpp/lib/MessageId.cc

```diff
@@ -30,43 +29,51 @@
 
 namespace pulsar {
 
-MessageId::MessageId() : ledgerId_(-1), entryId_(-1), partition_(-1) {}
+class MessageIdImpl {
+   public:
+    MessageIdImpl() : ledgerId_(-1), entryId_(-1), partition_(-1), batchIndex_(-1) {}
```

Comment:
Do we need this constructor 
I though in C++11 we can write 

const int64_t ledgerId_=-1

Reply:
But can we use c++11??? You guys tell me.. :)